### PR TITLE
feat: logging steps per second

### PR DESCRIPTION
**Pull request by sash-a**
## What?
Log the steps per second to tensorboard, neptune and sacred
## Why?
We were only printing the steps per second, would be nice to have it on neptune
## How?
Added steps per second to metrics dict
## Extra
* Removed `TimeIt` since we aren't using that anymore.
* Changed logging prefix to be e.g "trainer/" instead of "Trainer_"
* Added prefix for evalutor (it used to be blank)
* Changed `timesteps_per_training` to `steps_per_rollout`

### CODE REVIEW

File: mava/systems/ff_ippo_rware.py [hunk 1]
```diff
@@ -558,9 +559,11 @@ def run_experiment(_run: run.Run, _config: Dict, _log: SacredLogger) -> None:
...
 
         # Log the results of the evaluation.
+        elapsed_time = time.time() - start_time
+        evaluator_output.episodes_info["steps_per_second"] = steps_per_rollout / elapsed_time
         episode_return = log(
             metrics=evaluator_output,
-            t_env=timesteps_per_training * (i + 1),
+            t_env=steps_per_rollout * (i + 1),
         )
         if config["absolute_metric"] and max_episode_return <= episode_return:
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo_rware.py:568]: We need to add `evaluator_output.episodes_info["steps_per_second"] = steps_per_rollout / elapsed_time` to the absolute metric else it will cause an error of not finding steps_per_seconds within it

**Review by OmaymaMahjoub — CHANGES_REQUESTED** [review]
Looks good, just a minor fix related to logging the absolute metric!

File: mava/systems/ff_ippo_rware.py (hunk 1 above)
Comment (sash-a) [reply mava/systems/ff_ippo_rware.py:568]: Great spot!

**Review by RuanJohn — APPROVED** [review]
Very nice that we can log this now. Thanks @sash-a :fire:

File: mava/systems/ff_ippo_rware.py [hunk 2]
```diff
@@ -517,7 +517,7 @@ def run_experiment(_run: run.Run, _config: Dict, _log: SacredLogger) -> None:
     # Calculate total timesteps.
     n_devices = len(jax.devices())
     config["num_updates_per_eval"] = config["num_updates"] // config["num_evaluation"]
-    timesteps_per_training = (
+    steps_per_rollout = (
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo_rware.py:520]: It will be amazing if we can also print at the beginning of the training the expected total timesteps (as it is different depending on device num...)

File: mava/logger.py [hunk 3]
```diff
@@ -74,6 +75,8 @@ def log(
                 float(np.mean(episodes_length)),
                 t_env,
             )
+            logger.log_stat(prefix.lower() + "steps_per_second", steps_per_second, t_env)
+
             if trainer_metric:
```
Comment (OmaymaMahjoub) [inline mava/logger.py:80]: What do you think about adding a prefix to these info? I'm not sure if it would be beneficial.

**Review by OmaymaMahjoub — COMMENTED** [review]
Overall it looks good to me, Thanks for the PR! 
Maybe another minor request is to update the quickstart notebook to have `steps_per_rollout` instead of `timesteps_per_training`

File: mava/logger.py [hunk 4]
```diff
@@ -74,6 +75,8 @@ def log(
...
                 logger.log_stat("total_loss", float(np.mean(total_loss)), t_env)
                 logger.log_stat("value_loss", float(np.mean(value_loss)), t_env)
```
Comment (OmaymaMahjoub) [inline mava/logger.py:82]: What do you think about adding a prefix to these info? I'm not sure if it would be beneficial.

Comment (sash-a) [reply mava/logger.py:82]: What would the prefix be?

Comment (OmaymaMahjoub) [reply mava/logger.py:82]: maybe "trainer/" even though these only measured for the trainer :sweat_smile: ?

File: mava/systems/ff_ippo_rware.py (hunk 2 above)
Comment (OmaymaMahjoub) [reply mava/systems/ff_ippo_rware.py:529]: forget about this as it is solved on seperate config PR

**Review by OmaymaMahjoub — APPROVED** [review]
LGTM! Thanks @sash-a

File: mava/logger.py [hunk 5]
```diff
@@ -99,7 +94,8 @@ def log(
             f"Max Episode Return {float(np.max(episodes_return)):.3f} | "
             f"Mean Episode Length {float(np.mean(episodes_length)):.3f} | "
             f"Std Episode Length {float(np.std(episodes_length)):.3f} | "
-            f"Max Episode Length {float(np.max(episodes_length)):.3f}"
+            f"Max Episode Length {float(np.max(episodes_length)):.3f} | "
+            f"Steps Per Second {steps_per_second:.2e}"
```
Comment (RuanJohn) [inline mava/logger.py:98]:
```suggestion
            f"Steps Per Second {steps_per_second:.2e} "
```

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Thanks @sash-a, very minor suggestion. 
But also a detail, when I tried to run this locally the logging to Neptune seemed to not be working. Can you please double check that it works on your side?

**Review by RuanJohn — APPROVED** [review]
:fire:

**Review by OmaymaMahjoub — APPROVED** [review]
🥇
